src/privatesend-client.cpp

```diff
@@ -1002,13 +1002,24 @@ bool CPrivateSendClient::SubmitDenominate(CConnman& connman)
     std::vector<CTxOut> vecTxOutRet;
 
     // Submit transaction to the pool if we get here
-    // Try to use only inputs with the same number of rounds starting from the highest number of rounds possible
-    for(int i = nPrivateSendRounds; i > 0; i--) {
-        if(PrepareDenominate(i - 1, i, strError, vecTxDSInRet, vecTxOutRet)) {
-            LogPrintf("CPrivateSendClient::SubmitDenominate -- Running PrivateSend denominate for %d rounds, success\n", i);
-            return SendDenominate(vecTxDSInRet, vecTxOutRet, connman);
+    if (nLiquidityProvider) {
+        // Try to use only inputs with the same number of rounds starting from the lowest number of rounds possible
+        for(int i = 0; i< nPrivateSendRounds; i++) {
```

Comment:
Another C vs C++ style nit / chestnut* (chestnit?) - prefer pre inc/dec over post inc/dec.
https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement

Purely style here; i++ is traditional C (especially as a loop index like this) while ++i is idiomatic C++ (which therefore should've been called ++C).

When used in an expression a post-increment has to be read more carefully because it implies state.

* an old chestnut:
a joke or story that has become tedious because of its age and constant repetition.